### PR TITLE
AI Chat: Messaging

### PR DESCRIPTION
**Pull request by petemill**
Resolves https://github.com/brave/brave-browser/issues/32466 (privacy opt-in strings)
Resolves https://github.com/brave/brave-browser/issues/33894 (model strings)
Resolves https://github.com/brave/brave-browser/issues/33895

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/common/mojomUtils.ts [hunk 1]
```diff
@@ -32,3 +38,8 @@ export function mojoTimeToJSDate (mojoTime: mojo.Time): Date {
 export function mojoTimeDeltaToJSDate (mojoTime: mojo.TimeDelta) {
   return new Date(Number(mojoTime.microseconds) / 1000)
 }
+
+export function getKeysForMojomEnum(mojomEnum: MojomEnum): string[] {
```
Comment (fallaciousreasoning) [inline components/common/mojomUtils.ts:42]: Optionally, we could return the actual type strings of the enum:

```ts
export function getKeysForMojomEnum<T extends MojomEnum>(e: T): (Exclude<keyof T, 'MIN_VALUE' | 'MAX_VALUE'>)[] {
    const ignoreKeys = ['MIN_VALUE', 'MAX_VALUE']
    return Object.keys(e).filter(k => !ignoreKeys.includes(k)) as any
}
```

**Review by fallaciousreasoning — APPROVED** [review]
LGTM

File: components/common/mojomUtils.ts (hunk 1 above)
Comment (petemill) [reply components/common/mojomUtils.ts:42]: ooh that's nice

File: components/resources/ai_chat_ui_strings.grdp [hunk 2]
```diff
@@ -55,13 +58,13 @@
      Retry
   </message>
   <message name="IDS_CHAT_UI_INTRO_MESSAGE_CHAT_DEFAULT" desc="AI Chat intro message for the default model">
-    I'm here to help. What can I assist you with today?
+    Hi, I'm Leo. I'm a fully hosted AI assistant by Brave. I'm  powered by Llama 13B, a model created by Meta to be performant and applicable to many use cases.
   </message>
   <message name="IDS_CHAT_UI_INTRO_MESSAGE_CHAT_LEO_EXPANDED" desc="AI Chat intro message for the expanded model">
-    I have a vast base of knowledge and a large memory able to help with more complex challenges.
+    Hi, I'm Leo. I'm a fully hosted AI assistant by Brave.  I'm powered by Llama 70B, a model created by Meta to handle more advanced tasks than 13B.
```
Comment (mkarolin) [inline components/resources/ai_chat_ui_strings.grdp:64]: Super nit: there's an extra space between the sentences.

**Review by mkarolin — APPROVED** [review]
`strings`++

File: components/resources/ai_chat_ui_strings.grdp (hunk 2 above)
Comment (petemill) [reply components/resources/ai_chat_ui_strings.grdp:64]: that's not good, will fix. Thanks for spotting that!

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 11 x64` using the following build(s):

```
Brave | 1.62.2 Chromium: 119.0.6045.33 (Official Build) nightly (64-bit)
-- | --
Revision | 1ae4cbaeb4186df008e4562ed006f2c70a9beea8
OS | Windows 11 Version 22H2 (Build 22621.2428)
```

### `Test Case #1` - https://github.com/brave/brave-browser/issues/32466

Using the STR/Cases outlined via https://github.com/brave/brave-browser/issues/32466#issue-1862762752, ensured that the copy/message before using `Leo` matches the copy via https://docs.google.com/document/d/1DOj1RSSmZ82ER1aGYlFEkM1nmreT998Fe7bsKr1Y1T0/edit?pli=1#heading=h.lszc9i58xq4q as per the following:

image

### `Test Case #2` - https://github.com/brave/brave-browser/issues/33894

Using the Cases/Examples outlined via https://github.com/brave/brave-browser/issues/33894#issue-1960273897, ensured that the model copies are matching what's described via https://github.com/brave/brave-browser/issues/33894#issue-1960273897 as per the following:

**`Example`** | **`Example`** | **`Example`**
----------------|----------------|----------------
image | image | image | image

**Pull request merged**